src/scripts/browserDeviceProfile.js

```diff
@@ -1125,6 +1125,23 @@ export default function (options) {
         Conditions: h264CodecProfileConditions
     });
 
+    if (browser.web0s) {
+        if (supportsDolbyVision(options)) {
+            // Disallow direct playing of DOVI media in containers not mp4.
+            // This paired with the "Prefer fMP4-HLS Container" client playback setting enables DOVI playback on webOS.
+            profile.CodecProfiles.push({
+                Type: 'Video',
+                Container: "-mp4",
+                Codec: 'hevc',
+                Conditions: hevcCodecProfileConditions.map(condition =>
```

Comment:
Only necessary conditions can be left here - server checks all codec profiles that match codec+container.

I see that you haven't added `'mp4': 'SDR|HLG|HDR10|DOVI'`. But this is probably unnecessary:
- There will be `'-mp4': 'SDR|HLG|HDR10'` (matches non-MP4) and `'': 'SDR|HLG|HDR10|DOVI'` (matches any container).
- The first condition (`-mp4`) would fail for nonMP4+DoVi and the whole direct play test would fail (_as we want_).

Note: `'mp4': 'SDR|HLG|HDR10|DOVI'` + `'': 'SDR|HLG|HDR10'` won't work - specializations can only use a subset of the general profile.